packages/components/app/styles/components/code-block/index.scss

```
@@ -0,0 +1,209 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// CODE-BLOCK
+//
+
+
+// Note: "theme" contains just color variables and syntax highlighting styles
+@import "theme";
+
+// DIMENSIONS
+$code-block-line-numbers-width: 49px; // 3em = 49px
+$code-block-code-padding: 16px;
+
+// CODE-BLOCK PARENT/WRAPPER
+
+.hds-code-block {
+  position: relative;
+  overflow: hidden; // So header corners don't poke out from container
+  color: var(--code-block-color-foreground-primary);
+  font-family: var(--hds-typography-code-200);
+  background-color: var(--code-block-color-surface-primary);
+  // Note: may change border-radius to be toggled by an "@isRounded" property
+  border-radius: 6px;
+
+  pre,
+  code {
+    line-height: var(--token-typography-body-200-line-height); // = 1.4286 | 20px
+    white-space: pre;
+    text-align: left;
+    text-shadow: none;
+    word-wrap: normal;
+    word-break: normal;
+    word-spacing: normal;
+    tab-size: 4;
+    hyphens: none;
+
+    @media print { text-shadow: none; }
+  }
+}
+
+// CHILD COMPONENTS / ELEMENTS
+
+// Header (contains title & description)
+.hds-code-block__header {
+  display: flex;
+  flex-flow: column;
+  gap: 4px;
+  padding: 8px $code-block-code-padding;
+  background-color: var(--code-block-color-surface-faint);
+  border-bottom: 1px solid var(--code-block-color-border-strong);
```

Going through some of the feedback, I screwed up a little bit on the border styles, this should be `border-primary`
```suggestion
  border-bottom: 1px solid var(--code-block-color-border-primary);
```